File: backend/spring-boot-library/src/main/java/com/tai3/spring_boot_library/request/models/AddBookReq.java
package com.tai3.spring_boot_library.request.models;

import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
public class AddBookReq {
    String title;
    String author;
    String description;
    int copies;
    String category;
    String img;
}

File: frontend/react-library/src/models/AddBookReq.ts
class AddBookReq {
    constructor(
        public title: string,
        public author: string,
        public description: string,
        public copies: number,
        public category: string,
        public img?: string
    ) {
        this.title = title;
        this.author = author;
        this.description = description;
        this.copies = copies;
        this.category = category;
        this.img = img;
    }
}

export default AddBookReq;